tgui/packages/tgui/interfaces/ListInput.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * @file
+ * @copyright 2020 watermelon914 (https://github.com/watermelon914)
+ * @license MIT
+ */
+
+import { clamp01 } from 'common/math';
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Flex, Section, Input } from '../components';
+import { Window } from '../layouts';
+
+const ARROW_KEY_UP = 38;
+const ARROW_KEY_DOWN = 40;
+
+let lastScrollTime = 0;
+
+export const ListInput = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    title,
+    message,
+    buttons,
+    timeout,
+  } = data;
+
+  // Search
+  const [showSearchBar, setShowSearchBar] = useLocalState(
+    context, 'search_bar', false);
+  const [displayedArray, setDisplayedArray] = useLocalState(
+    context, 'displayed_array', buttons);
+
+  // KeyPress
+  const [searchArray, setSearchArray] = useLocalState(
+    context, 'search_array', []);
+  const [searchIndex, setSearchIndex] = useLocalState(
+    context, 'search_index', 0);
+  const [lastCharCode, setLastCharCode] = useLocalState(
+    context, 'last_char_code', null);
+
+  // Selected Button
+  const [selectedButton, setSelectedButton] = useLocalState(
+    context, 'selected_button', buttons[0]);
+  return (
+    <Window
+      title={title}
+      width={325}
+      height={325}
+      resizable
+    >
+      {timeout !== undefined && <Loader value={timeout} />}
+      <Window.Content>
+        <Flex direction="column" height="100%">
+          <Flex.Item grow={1}>
+            <Section
+              scrollable
+              className="ListInput__Section"
+              width="100%"
+              fill
+              title={message}
+              tabIndex={0}
+              onKeyDown={e => {
+                e.preventDefault();
+                if (lastScrollTime > performance.now()) {
+                  return;
+                }
+                lastScrollTime = performance.now() + 125;
+
+                if (e.keyCode === ARROW_KEY_UP || e.keyCode === ARROW_KEY_DOWN)
+                {
+                  let direction = 1;
+                  if (e.keyCode === ARROW_KEY_UP) direction = -1;
+
+                  let index = 0;
+                  for (index; index < buttons.length; index++) {
+                    if (buttons[index] === selectedButton) break;
+                  }
+                  index += direction;
+                  if (index < 0) index = buttons.length-1;
+                  else if (index >= buttons.length) index = 0;
+                  setSelectedButton(buttons[index]);
+                  setLastCharCode(null);
+                  document.getElementById(buttons[index]).focus();
+                  return;
```

Comment:
```suggestion
                  let index = buttons.findIndex(selectedButton);
                  if (index === -1) {
                    index = 0;
                  }
                  index = (index + direction) % buttons.length;
                  const button = buttons[index];
                  setSelectedButton(button);
                  setLastCharCode(null);
                  document.getElementById(button).focus();
```
I know there's 100 ways to write this, but the one as-written has some *weird* patterns in it. Only thing that would need confirming is whether `findIndex` is available in the polyfill, as that's not in non-Edge IE.

Reply:
looks like it, at least in the bundle

Reply:
didn't add the removal of `return`